storage/deliveryjobrepo.go

```diff
@@ -192,6 +206,17 @@ func (djRepo *DeliveryJobDBRepository) GetJobsForConsumer(consumer *data.Consume
 	return djRepo.getJobs(baseQuery, nil, consumer, appendWithPaginationArgs(page, consumer.ID.String(), jobStatus))
 }
 
+// GetPrioritizedJobsForConsumer retrieves DeliveryJob created for delivery to a customer and it has to be filtered by a specific status and ordered by message priority
+func (djRepo *DeliveryJobDBRepository) GetPrioritizedJobsForConsumer(consumer *data.Consumer, jobStatus data.JobStatus, pageSize int) ([]*data.DeliveryJob, error) {
+	fields := "id, messageId, consumerId, status, dispatchReceivedAt, retryAttemptCount, statusChangedAt, earliestNextAttemptAt, createdAt, updatedAt"
+	joinedTables := "job JOIN (SELECT id as mid, priority FROM message) as M ON messageId=M.mid"
```

Comment:
This is a interesting way to join; but since its an inner SQL not sure if there is a performance penalty of traditional straight forward join.
@imran2140 bhai do you have any idea?

Reply:
This query can be potentially extremely costly given the volume of message and jobs; we should even consider deduplicating priority to job to avoid this join. What do you think @imran2140 bhai?

Reply:
Agreed, the subquery result is unbounded.  Also, we're ordering by fields that are are not in `WHERE` condition, so MySQL is likely to use temporary tables / `filesort` to return results.  For optimal performance, the WHERE and ORDER BY clauses should be covered by the same index.

Reply:
The subquery was necessary to stay compatible with existing cursor based pagination. Since we're not using pagination now, I've removed the subquery part: https://github.com/newscred/webhook-broker/pull/141/files#diff-ba506b93dfd1bad65e6bcbbbe5f62d2a2015292a903eb9e5b71aa8a9750f1e25R211-R215